core/src/ldml/ldml_transforms.cpp

```diff
@@ -420,29 +412,35 @@ reorder_group::apply(std::u32string &str) const {
   }
   // recombine into a string by pulling out the 'ch' value
   // that's in each sortkey element.
-  std::u32string newSuffix;
+  out.clear(); // will re-add all text
   signed char q = sort_keys.begin()->quaternary; // start with the first quaternary
   for (auto e = sort_keys.begin(); e < sort_keys.end(); e++, q++) {
     if (q != e->quaternary) {
       // something rearranged in this subrange, because the quaternary values are out of order.
       applied = true;
     }
     // collect the characters
-    newSuffix.append(1, e->ch);
+    out.append(1, e->ch);
   }
-  if (applied) {
-    str.resize(prefix.size());
-    str.append(newSuffix);
-  } else {
+  if (!applied) {
     DebugTran("Skip: sorting caused no reordering");
+    // exit early to avoid string copying and possibly marker re-adding.
+    return false; // no change
   }
 #if KMXPLUS_DEBUG_TRANSFORM
   DebugTran("Sorted sortkey");
   for (const auto &r : sort_keys) {
     r.dump();
   }
 #endif
-  return applied;
+  if (markers.empty()) {
+    // no markers, just copy over to str
+    str = out;
+  } else {
+    // copy and re-add markers
+    add_back_markers(str, out, markers, plain_sentinel);
+  }
```

Comment:
I think it'd be better to see this logic inside `add_back_markers()` -- simplifies usage patterns for it. It should have no effective performance difference.

Reply:
yes. need to make an update downstream for this too, but added